### PR TITLE
fix(DB): Quest "In Dreams"

### PR DESCRIPTION
**Pull request by Eronox**
## Changes Proposed:
-  Port from https://gist.githubusercontent.com/Jinnaix/347ecc74b3e546c608fe465dd3fcb415/raw/b42f7dc1a0f118295629a628c36d73752c1cfc0c/InDreams.sql
-  Minor adjustments to make it follow azerothcore guideline
-  Minor adjustments to make it more blizzlike

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8345
- Closes https://github.com/chromiecraft/chromiecraft/issues/2014

## SOURCE:

Video: https://www.youtube.com/watch?v=QBHpy8an_p4&ab_channel=Bue
Note: Taelan, even though mounted, seems to not be ever going in a run speed. I thought this was weird, but can't find anything proving that he should ride in run speed. Hence the sloooooow quest progression

## Tests Performed:

- Ran it through ingame, could not identify major issues
- Query re-appliable

## How to Test the Changes:

```
-- Step: 1
.cast 17961
.que add 5862
.go crea 45436

-- Step: 2
Turn in quest 
Take new quest
.unaura 17961 (when Taelan says `Remove your disguise`)

-- Step:3 
Follow Taelan, kill any attacking creature. 
Outside tower, when you meet ISILLIEN, don't kill anything just stay with `.cheat god on` and compare to video below
```

## Known Issues and TODO List:

- End RP with Tirion over Taelans dead body is not flawless
- No safety net included, making sure Taelan dies _(but should always die)_

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1640818723105671700.sql [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+(@TAELAN, 70, 2696.58, -1856.38, 68.2104, 'Taelan Fordring'),
+(@TAELAN, 71, 2694.06, -1870.35, 66.9045, 'Taelan Fordring'),
+(@TAELAN, 72, 2693.26, -1873.62, 66.8413, 'Taelan Fordring'),
+(@TAELAN, 73, 2675.36, -1891.94, 66.1742, 'Taelan Fordring'),
+(@TAELAN, 74, 2669.33, -1898.11, 66.7004, 'Taelan Fordring');
+
+-- Conditions
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=@DEATHSPELL;
+INSERT INTO `conditions` VALUES
+(13, 1, @DEATHSPELL, 0, 0, 31, 0, 3, @TAELAN, 0, 0, 0, 0, '', 'Taelan Death Spell - Target Taelan');
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1640818723105671700.sql:269]:
```suggestion
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(13, 1, @DEATHSPELL, 0, 0, 31, 0, 3, @TAELAN, 0, 0, 0, 0, '', 'Taelan Death Spell - Target Taelan');
```

File: data/sql/updates/pending_db_world/rev_1640818723105671700.sql [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+(@TIRION,6,0,'Too long have I sat idle, gripped in this haze... this malaise, lamenting what could have been... what should have been.',12,0,100,0,0,0,7429,0,'Lord Tirion Fordring'),
+(@TIRION,7,0,'Your death will not have been in vain, Taelan. A new Order is born on this day... an Order which will dedicate itself to extinguishing the evil that plagues this world. An evil that cannot hide behind politics and pleasantries.',12,0,100,0,0,0,7426,0,'Lord Tirion Fordring'),
+(@TIRION,8,0,'This I promise... This I vow...',12,0,100,0,0,0,7430,0,'Lord Tirion Fordring');
+
+-- Grand Inquisitor Isillien
+UPDATE `creature_template` SET `faction`=16, `AIName`='SmartAI', `ScriptName`='' WHERE entry=@ISILLIEN;
+
+DELETE FROM smart_scripts WHERE entryOrGuid = @ISILLIEN AND source_type = 0;
+INSERT INTO smart_scripts (entryorguid, source_type, id, link, event_type, event_phase_mask, event_chance, event_flags, event_param1, event_param2, event_param3, event_param4, action_type, action_param1, action_param2, action_param3, action_param4, action_param5, action_param6, target_type, target_param1, target_param2, target_param3, target_x, target_y, target_z, target_o, comment) VALUES
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1640818723105671700.sql:50]:
```suggestion
INSERT INTO smart_scripts (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
```

File: data/sql/updates/pending_db_world/rev_1640818723105671700.sql [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+(@TIRION,4,0,'%s falls to one knee.',16,0,100,0,0,0,7421,0,'Lord Tirion Fordring'),
+(@TIRION,5,0,'Look what they did to my boy.',12,0,100,5,0,0,7423,0,'Lord Tirion Fordring'),
+(@TIRION,6,0,'Too long have I sat idle, gripped in this haze... this malaise, lamenting what could have been... what should have been.',12,0,100,0,0,0,7429,0,'Lord Tirion Fordring'),
+(@TIRION,7,0,'Your death will not have been in vain, Taelan. A new Order is born on this day... an Order which will dedicate itself to extinguishing the evil that plagues this world. An evil that cannot hide behind politics and pleasantries.',12,0,100,0,0,0,7426,0,'Lord Tirion Fordring'),
+(@TIRION,8,0,'This I promise... This I vow...',12,0,100,0,0,0,7430,0,'Lord Tirion Fordring');
+
+-- Grand Inquisitor Isillien
+UPDATE `creature_template` SET `faction`=16, `AIName`='SmartAI', `ScriptName`='' WHERE entry=@ISILLIEN;
+
+DELETE FROM smart_scripts WHERE entryOrGuid = @ISILLIEN AND source_type = 0;
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1640818723105671700.sql:49]:
```suggestion
DELETE FROM `smart_scripts` WHERE `entryOrGuid` = @ISILLIEN AND `source_type` = 0;
```

File: data/sql/updates/pending_db_world/rev_1640818723105671700.sql [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+(@TIRION,2,0,'Face me, coward. Face the faith and strength that you once embodied.',12,0,100,25,0,0,7374,0,'Lord Tirion Fordring'),
+(@TIRION,3,0,'A thousand more like him exist. Ten thousand. Should one fall, another will rise to take the seat of power.',12,0,100,0,0,0,7420,0,'Lord Tirion Fordring'),
+(@TIRION,4,0,'%s falls to one knee.',16,0,100,0,0,0,7421,0,'Lord Tirion Fordring'),
+(@TIRION,5,0,'Look what they did to my boy.',12,0,100,5,0,0,7423,0,'Lord Tirion Fordring'),
+(@TIRION,6,0,'Too long have I sat idle, gripped in this haze... this malaise, lamenting what could have been... what should have been.',12,0,100,0,0,0,7429,0,'Lord Tirion Fordring'),
+(@TIRION,7,0,'Your death will not have been in vain, Taelan. A new Order is born on this day... an Order which will dedicate itself to extinguishing the evil that plagues this world. An evil that cannot hide behind politics and pleasantries.',12,0,100,0,0,0,7426,0,'Lord Tirion Fordring'),
+(@TIRION,8,0,'This I promise... This I vow...',12,0,100,0,0,0,7430,0,'Lord Tirion Fordring');
+
+-- Grand Inquisitor Isillien
+UPDATE `creature_template` SET `faction`=16, `AIName`='SmartAI', `ScriptName`='' WHERE entry=@ISILLIEN;
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1640818723105671700.sql:47]:
```suggestion
UPDATE `creature_template` SET `faction`=16, `AIName`='SmartAI', `ScriptName`='' WHERE `entry`=@ISILLIEN;
```

**Comment by Eronox** [comment]
Thanks for fixing ticks (`) @Nyeriah

File: data/sql/updates/pending_db_world/rev_1640818723105671700.sql [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+(@TIRION,7,0,'Your death will not have been in vain, Taelan. A new Order is born on this day... an Order which will dedicate itself to extinguishing the evil that plagues this world. An evil that cannot hide behind politics and pleasantries.',12,0,100,0,0,0,7426,0,'Lord Tirion Fordring'),
+(@TIRION,8,0,'This I promise... This I vow...',12,0,100,0,0,0,7430,0,'Lord Tirion Fordring');
+
+-- Grand Inquisitor Isillien
+UPDATE `creature_template` SET `faction`=16, `AIName`='SmartAI', `ScriptName`='' WHERE `entry`=@ISILLIEN;
+
+DELETE FROM `smart_scripts` WHERE `entryOrGuid` = @ISILLIEN AND `source_type` = 0;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1640818723105671700.sql:50]:
```suggestion
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
```

**Comment by Nyeriah** [comment]
Tested, working as intended, thanks for the PR!

**Pull request merged**